### PR TITLE
Adds fixes to @WP/block-editor

### PR DESCRIPTION
**Pull request by margolisj**
Adds some missing props to block editor package:
useInnerBlockProps:
orientation - https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/nested-blocks-inner-blocks/#orientation

BlockControls:
groups - https://github.com/WordPress/gutenberg/blob/43169696cbbfb6fce623f6fd3e02cdd1921e63b2/packages/block-editor/src/components/block-controls/groups.js#L12

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/wordpress__block-editor/components/block-controls.d.ts
1. types/wordpress__block-editor/components/inner-blocks.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d7418b431437be2d2b6b7035cacfdece198fa70e
</p>

File: types/wordpress__block-editor/components/block-controls.d.ts [hunk 1]
```diff
@@ -2,8 +2,16 @@ import { Slot, Toolbar } from "@wordpress/components";
...
+    type BlockControlGroup =
+        | "default"
+        | "block"
+        | "inline"
+        | "other"
+        | "parent";
+
     interface Props extends Pick<Toolbar.Props, "controls"> {
         children: ReactNode;
+        group?: BlockControlGroup;
```
Comment (MysteryBlokHed) [inline types/wordpress__block-editor/components/block-controls.d.ts:14]: I wouldn't call this a requirement, but you can change your optionals to use `| undefined` for the `exactOptionalPropertyTypes` config option: <https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes>

```suggestion
        group?: BlockControlGroup | undefined;
```

(This applies to other optional properties, not just this one)

**Review by MysteryBlokHed — COMMENTED** [review]
If you update the tests ([`wordpress__block-editor-tests.tsx`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wordpress__block-editor/wordpress__block-editor-tests.tsx)), then this can be merged without needing a DefintelyTyped maintainer's  review.

File: types/wordpress__block-editor/components/block-controls.d.ts (hunk 1 above)
Comment (margolisj) [reply types/wordpress__block-editor/components/block-controls.d.ts:14]: Yeah happy to. Any reason why? I thought `?` added `undefined` to the set. Not pushing back on the req, just curious. Thanks!

**Comment by margolisj** [comment]
> If you update the tests ([`wordpress__block-editor-tests.tsx`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wordpress__block-editor/wordpress__block-editor-tests.tsx)), then this can be merged without needing a DefintelyTyped maintainer's review.

Will do. Thanks!

File: types/wordpress__block-editor/components/block-controls.d.ts (hunk 1 above)
Comment (MysteryBlokHed) [reply types/wordpress__block-editor/components/block-controls.d.ts:14]: There are a few cases in JavaScript where there's actually a difference between explicitly passing `undefined`, and just not providing a value at all. You can take a look at the page I linked if you're interested in reading a bit more about it: <https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes>

Comment (margolisj) [reply types/wordpress__block-editor/components/block-controls.d.ts]: Appreciate it.

**Review by Anonymous4078 — APPROVED** [review]

**Comment by margolisj** [comment]
Ready to merge.

**Pull request merged**